src/draw/shapes.typ

```diff
@@ -312,20 +312,27 @@
       path: drawables
     )
 
+    let transformed = false
     if mark_.check-mark(style.mark) {
+      if style.mark.post-transform {
+        drawables = drawable.apply-transform(transform, drawables).first()
+        transformed = true
+      }
+
       let (marks, segments) = mark_.place-marks-along-path(ctx, style.mark, drawables.segments)
       drawables.segments = segments
       drawables = (drawables,) +  marks
     }
 
+    if not transformed {
+      drawables = drawable.apply-transform(transform, drawables)
+    }
+
     return (
       ctx: ctx,
       name: name,
       anchors: anchors,
-      drawables: drawable.apply-transform(
-        transform,
-        drawables,
-      )
+      drawables: drawables,
```

Comment:
I think this implementation could be improved. Why not transform the drawables inside `place-marks`, then have transform the drawable if no marks are placed.

Although with this change I'm not sure about how the future of modifiers could be implemented for marks...

Reply:
As mentioned in the PR text, I do not think this is optimal. If the marks do the transform it is a bit too "hidden" and it won't save that many lines either. I'd rather have it done before calling `place-marks-along-path`.

Reply:
I disagree that the transformation would be too "hidden", the `transform` argument would be explicitly passed and would always happen. I think having and repeating some code that affects marks outside of a mark function is not good.